rules/common.smk

```diff
@@ -1,17 +1,45 @@
 import pandas as pd
+from peppy import Project, SAMPLE_NAME_COLNAME as PEP_SAMPLE_COL
 from snakemake.utils import validate
 
-report: "../report/workflow.rst"
+
+def peppy_rename(df):
+    """ Rename peppy's column for sample name identification to snakemake's. """
+    if df is None:
+        return None
+    if "sample" in df.columns and PEP_SAMPLE_COL in df.columns:
+        raise Exception("Multiple sample identifier columns present: {}".format(", ".join(["sample", PEP_SAMPLE_COL])))
+    return df.rename({PEP_SAMPLE_COL: "sample"}, axis=1)
+
+
+def peppy_units(df):
+    """ Add unit/subsample indices to peppy a data frame.  """
+    if "unit" in df.columns:
+        return df
+    def count_names(names):
+        def go(rem, n, curr, acc):
+            if rem == []:
+                return acc + [n]
+            h, t = rem[0], rem[1:]
+            return go(t, n + 1, curr, acc) if h == curr else go(t, 1, h, acc + [n])
+        return go(names[1:], 1, names[0], []) if names else []
+    df.insert(1, "unit", [i for n in count_names(list(df[PEP_SAMPLE_COL])) for i in range(1, n + 1)])
+    return df
+
 
 ###### Config file and sample sheets #####
-configfile: "config.yaml"
+p = Project("prjcfg.yaml")
+configfile: p.snake_config
 validate(config, schema="../schemas/config.schema.yaml")
 
-samples = pd.read_table(config["samples"]).set_index("sample", drop=False)
+samples = p.sheet
+samples = peppy_rename(samples).set_index("sample", drop=False)
+
 validate(samples, schema="../schemas/samples.schema.yaml")
 
-units = pd.read_table(config["units"], dtype=str).set_index(["sample", "unit"], drop=False)
+units = peppy_units(peppy_rename(p.sample_subannotation)).set_index(["sample", "unit"], drop=False)
```

Comment:
I would prefer if peppy would set the dataframe index like that itself. Is there a reason to not do it?

Reply:
That should already be fine, or if not we can probably accommodate it

Reply:
Does peppy know about the unit column at all? I thought it just requires the sample_name column in the subannotation.

Reply:
we added it above to accommodate snakemake... but in our typical use case, the `unit` column is called `subsample_name` -- but it is optional in generic `peppy` Projects...